sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -94,6 +94,28 @@ class GpuHashAggregateMeta(
             s"'final', 'complete', or 'all'")
       }
     }
+    hashAggReplaceMode match {
+      case "partial" => {
+        // In partial mode, if there are non distinct functions and multiple distinct functions
+        // non distinct functions are computed using First operator. The final result would be
+        // incorrect for non distinct functions for partition size > 1. Falling back to CPU for
+        // this special case.
+        if (agg.aggregateExpressions.exists(e => e.aggregateFunction.isInstanceOf[First])) {
+          val count = agg.aggregateExpressions.flatMap(expr =>
+            expr.children.flatMap {
+              _.collect {
+              case a: Count => a
+            }
+          })
+          if (count.size > 1) {
+            willNotWorkOnGpu("Aggregate of non distinct functions with multiple distinct " +
+              "functions is non deterministic for non distinct functions as it is computed using " +
+              "First.")
```

Comment:
Have we verified that this bug will not happen with average distinct with other non-distinct aggregates?

Reply:
To be on the safer side, having those tests in scala/ python would be very useful.

Reply:
@nartal1 , hope this helps, Sample test that is causing failures on returned avgs. I see First there and that seems a related manifestation to me.
```
  IGNORE_ORDER_ALLOW_NON_GPU_testSparkResultsAreEqual(
    "Agg expression with filter with nulls", longsCsvDf, execsAllowedNonGpu =
    Seq("HashAggregateExec", "AggregateExpression", "AttributeReference", "Alias", "Average",
      "Cast", "If", "EqualTo", "Literal", "First"),
    conf = partialOnlyConf, repart = 2) {
    frame => frame.createOrReplaceTempView("testTable")
      val res = frame.sparkSession.sql(
        s"""
           | SELECT
           |   avg(distinct more_longs),
           |   avg(more_longs),
           |   avg(distinct longs)
           | FROM testTable
           |   group by longs
           |""".stripMargin)
      res.explain()
      res
  }
```

Reply:
Thanks, Kuhu. I updated the code and added this test in python tests. Please take another look